client/data/charges/types.ts

```diff
@@ -0,0 +1,172 @@
+/* eslint-disable camelcase */
+export interface Address {
```

Comment:
How about setting all fields as required but use [Parital utility](https://www.typescriptlang.org/docs/handbook/utility-types.html#partialtype)?

Reply:
I think the `Partial<T>` type is really only useful for function declarations. The issue here is that Stripe (or even our API) may return `null` for each of these fields which is why they're optional.

The type interface here doesn't have anything to do with using the type, it's more about defining the structure of the type. The `Partial<T>` utility is intended for when you need to use a variable that might only contain partial information for a type `T`.

Reply:
Fair point. Depending on the outcome of conversation in https://github.com/Automattic/woocommerce-payments/pull/1279#discussion_r577477482 we might still end up with something like `interface WCPayType = Partial<StripeType>`.